src/Hazelcast.Net/Sql/SqlColumnType.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Hazelcast.Sql
+{
+    /// <summary>
+    /// SQL column type.
+    /// </summary>
+    public enum SqlColumnType
+    {
+        /// <summary>
+        /// <c>VARCHAR</c> type, represented by <c>java.lang.String</c> in Java and by <see cref="string"/> in .NET.
+        /// </summary>
+        Varchar = 0,
+
+        /// <summary>
+        /// <c>BOOLEAN</c> type, represented by <c>java.lang.Boolean</c> in Java and by <see cref="bool"/> in .NET.
+        /// </summary>
+        Boolean = 1,
+
+        // FIXME [Oleksii] discuss sign preservation
+        /// <summary>
+        /// <c>TINYINT</c> type, represented by <c>java.lang.Byte</c> in Java and by <see cref="byte"/> in .NET.
+        /// </summary>
+        TinyInt = 2,
+
+        /// <summary>
+        /// <c>SMALLINT</c> type, represented by <c>java.lang.Short</c> in Java and by <see cref="short"/> in .NET.
+        /// </summary>
+        SmallInt = 3,
+
+        /// <summary>
+        /// <c>INTEGER</c> type, represented by <c>java.lang.Integer</c> in Java and by <see cref="int"/> in .NET.
+        /// </summary>
+        Integer = 4,
+
+        /// <summary>
+        /// <c>BIGINT</c> type, represented by <c>java.lang.Long</c> in Java and by <see cref="long"/> in .NET.
+        /// </summary>
+        BigInt = 5,
+
+        /// <summary>
+        /// <c>DECIMAL</c> type, represented by <c>java.lang.BigDecimal</c> in Java and by <see cref="string"/> in .NET.
+        /// </summary>
+        Decimal = 6,
+
+        /// <summary>
+        /// <c>REAL</c> type, represented by <c>java.lang.Float</c> in Java and by <see cref="float"/> in .NET.
+        /// </summary>
+        Real = 7,
+
+        /// <summary>
+        /// <c>DOUBLE</c> type, represented by <c>java.lang.Double</c> in Java and by <see cref="double"/> in .NET.
+        /// </summary>
+        Double = 8,
+
+        // FIXME [Oleksii] discuss year range in HZ SQL and Java
```

Comment:
@mdumandag 
According to [docs](https://docs.hazelcast.com/imdg/4.2/sql/data-types.html) `DATE` type is backed by `java.lang.LocalDate` which supports [year range of ±999,999,999](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html#MIN). However, it's serialized as short (2 bytes) so real supported values ranges from -32768 to 32767.

For now I limited custom `HLocalDate` with second option to avoid data loss, but this probably need to be clarified in .NET and maybe IMDG documentation.

Reply:
Looks like this was fixed in one of the updates and it's serialized as int (4 bytes) everywhere.